underdocs-error/src/test/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinterTest.kt

```diff
@@ -0,0 +1,64 @@
+package underdocs.error.pretty
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import underdocs.error.BaseUnderdocsError
+
+class DefaultErrorPrettyPrinterTest {
+
+    companion object {
+        const val TITLE = "Test Exception"
+        const val ERROR_CODE = "42"
+        const val DESCRIPTION = "This is a dummy exception for tests."
+        const val SOLUTION = "There is no solution for this."
+        const val RUNTIME_EXCEPTION_CLASSPATH = "java.lang.RuntimeException"
+        const val RUNTIME_EXCEPTION_DESCRIPTION = "RuntimeException Description"
+    }
+
+    private val defaultErrorPrettyPrinter = DefaultErrorPrettyPrinter()
+
+    @Test
+    fun print_WhenBaseUnderDocsErrorTypeExceptionIsThrown_ShouldReturnMessageContainingTheDetailsOfTheException() {
```

Comment:
The title of this test is great, however, it reads awful as a function name. Fortunately, both in Java and Kotlin, we can do better!

**Kotlin**

Backtick function names to the rescue:

~~~~Kotlin
@Test
fun `print | When BaseUnderdocsError type exception is thrown, Should return message containing the details of the exception`() {
}
~~~~

Link: https://kotlinlang.org/docs/reference/coding-conventions.html#names-for-test-methods

From time to time, it would not jump to the appropriate section. If that occurs, simply search for "Names for Test Methods" on the page.

**Java**

One can leverage JUnit's `@DisplayName` annotation:

~~~~Java
@DisplayName("print | When BaseUnderdocsError type exception is thrown, Should return message containing the details of the exception")
@Test
void print_WhenBaseUnderDocsErrorTypeExceptionIsThrown_ShouldReturnMessageContainingTheDetailsOfTheException() {
}
~~~~

Link: https://junit.org/junit5/docs/current/user-guide/#writing-tests-display-names